src/lib/viewers/media/MediaBaseViewer.js

```diff
@@ -122,16 +142,15 @@ class MediaBaseViewer extends BaseViewer {
         // Try catch is needed due to weird behavior when src is removed
         try {
             if (this.mediaEl) {
-                this.mediaEl.removeEventListener('timeupdate', this.setTimeCode);
-                this.mediaEl.removeEventListener('progress', this.progressHandler);
-                this.mediaEl.removeEventListener('volumechange', this.updateVolumeIcon);
-                this.mediaEl.removeEventListener('playing', this.playingHandler);
-                this.mediaEl.removeEventListener('pause', this.pauseHandler);
-                this.mediaEl.removeEventListener('ended', this.resetPlayIcon);
-                this.mediaEl.removeEventListener('seeked', this.seekHandler);
+                this.removeEventListenersForMediaElement();
+
                 this.mediaEl.removeEventListener('loadeddata', this.loadeddataHandler);
                 this.mediaEl.removeEventListener('error', this.errorHandler);
 
+                // Attempt to process the playback metrics at whatever point of playback has occurred
+                // before we destroy the viewer
+                this.processMetrics();
```

Comment:
Should we do this at the very top of `destroy`? Also, is it ever going to be possible for metrics to be emitted twice, once by `beforeunload` and once here?

Reply:
yeah I can do that, but I think what I'll do is also add a guard clause in `processMetrics` to check `this.loaded` to make sure data has loaded first instead of emitting metrics in cases where the viewer hasn't even loaded any data

Also, yeah it should only be emitted once since it's whitelisted in `getMetricsWhitelist`